src/matrixeventprocessor.ts

```diff
@@ -106,38 +119,32 @@ export class MatrixEventProcessor {
         } else if (["m.room.member", "m.room.name", "m.room.topic"].includes(event.type)) {
             await this.ProcessStateEvent(event);
             return;
-        } else if (event.type === "m.room.redaction" && context.rooms.remote) {
+        } else if (event.type === "m.room.redaction") {
+            if (!context.rooms.remote) {
+                log.info("Got radaction event with no linked room. Ignoring.");
+                return;
+            }
             await this.discord.ProcessMatrixRedact(event);
             return;
         } else if (event.type === "m.room.message" || event.type === "m.sticker") {
             log.verbose(`Got ${event.type} event`);
-            const isBotCommand = event.type === "m.room.message" &&
-                event.content!.body &&
-                event.content!.body!.startsWith("!discord");
-            if (isBotCommand) {
+            if (isBotCommand(event)) {
                 await this.mxCommandHandler.Process(event, context);
-                return;
-            } else if (context.rooms.remote) {
-                const srvChanPair = context.rooms.remote.roomId.substr("_discord".length).split("_", ROOM_NAME_PARTS);
-                try {
-                    await this.ProcessMsgEvent(event, srvChanPair[0], srvChanPair[1]);
-                    return;
-                } catch (err) {
-                    log.warn("There was an error sending a matrix event", err);
-                    return;
-                }
+            } else {
+                await this.ProcessMsgEvent(event, context);
             }
+            return;
         } else if (event.type === "m.room.encryption" && context.rooms.remote) {
             try {
                 await this.HandleEncryptionWarning(event.room_id);
                 return;
             } catch (err) {
-                throw new Error(`Failed to handle encrypted room, ${err}`);
+                throw wrap(err, unstable.EventNotHandledError, `Failed to handle encrypted room, ${err}`);
             }
         } else {
-            log.verbose("Got non m.room.message event");
+            throw new unstable.EventUnknownError("Got non m.room.message event");
```

Comment:
This is the most exciting bridge error to try out in the wild and one that is quite useful for the user in the end.

I think it will fire for too many events initially and it might become spammy. My current idea is to special case the most common events in a room for which it would fire (especially non-messaging related events), so that it is not annoying, but still catches unknown stuff. (I remember writing that as a more articulated answer to Travis over at the MSC, but I didn't find it at first look…)

I also don't think that it might DoS the bridge, if that is your concern. In contrast to the discussion we had about `m.event_too_old` where events accumulated over many hours might flood in at once, this one will at most spawn as often as the number of incoming events.

Reply:
Hmm, fair point. I agree it's probably not world ending.